lib/blocktree/blocktree.go

```diff
@@ -68,6 +69,28 @@ func (bt *BlockTree) GenesisHash() Hash {
 	return bt.head.hash
 }
 
+// VerifyBlockExists inserts the block as child of its parent node
+// Note: Assumes block has no children
+func (bt *BlockTree) VerifyHeaderValid(header *types.Header) error {
+	genesisHashStr := header.ParentHash.String()
+	log.Warn("VerifyHeaderValid, genesisHashStr,", genesisHashStr)
+
+	parent := bt.getNode(header.ParentHash)
+	if parent == nil {
+		err := fmt.Errorf("cannot find parent block in blocktree")
+		log.Error("Could not getNode for ParentHash", "header.ParentHash", header.ParentHash, "err", err)
+		return err
+	}
+
+	// Check if it already exists
+	n := bt.getNode(header.Hash())
+	if n != nil {
+		return fmt.Errorf("cannot add block to blocktree that already exists: hash=%s", n.hash)
+	}
+
+	return nil
```

Comment:
this seems to be a duplicate of blocktree.AddNode? 

also, you shouldn't add the block to the blocktree when verifying, and it's not actually possible to complete the verification from the blocktree, since you need the block digests. change this to a function that calculates the depth of the block you want to add with `bt.getNode(parentHash).depth` and return the hashes of all other nodes with the same depth

Reply:
I addressed the point mentioned, still I am not sure if this is correct. Could you please review it again ? :)